core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -706,6 +706,8 @@ abstract class AbstractFetcherThread(name: String,
    *    - the request succeeded or
    *    - it was fenced and this thread hasn't received new epoch, which means we need not backoff and retry as the
    *    partition is moved to failed state.
+   *    - for ReplicaAlterLogDirsThread, "OutOfRange" error is generated when starting fetch to leader (source dir,
```

Comment:
Just for my understanding, the new replica starts with trying to fetch offset 0, and in the case 0 is not the first offset, throws OffsetOutOfRange?

Reply:
I believe a better way to explain the behaviour is that the OffsetOutOfRange error is used to ask the new replica (B) to set the offset it wants to fetch from to the earliest one available from A, but I will double-check my claim

Reply:
That's another way of phrasing my question -- thanks for checking :)